From 02ae61e9c620ec8bff1a99fde4b68205c346d811 Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Wed, 17 Apr 2019 18:30:39 +0200
Subject: [PATCH] board: tqma8qxs: select default u-boot and linux device tree
 files from Kconfig

while at it, fix KConfig statement for MB-SMARC-2: no lower case in Kconfig symbols

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 board/tqc/tqma8qxs/Kconfig  | 12 ++++++++++--
 board/tqc/tqma8qxs/Makefile |  2 +-
 include/configs/tqma8qxs.h  |  2 +-
 3 files changed, 12 insertions(+), 4 deletions(-)

diff --git a/board/tqc/tqma8qxs/Kconfig b/board/tqc/tqma8qxs/Kconfig
index 1d3035aa86..08ff2340fb 100644
--- a/board/tqc/tqma8qxs/Kconfig
+++ b/board/tqc/tqma8qxs/Kconfig
@@ -30,11 +30,11 @@ endchoice
 
 choice
 	prompt "TQMa8QXS base board variant"
-	default TQMa8QXS_MB_SMARC_2
+	default TQMA8QXS_MB_SMARC_2
 	help
 	  Select base board for TQMa8QXS
 
-config TQMa8QXS_MB_SMARC_2
+config TQMA8QXS_MB_SMARC_2
 	bool "TQMa8QXS on MB_SMARC_2 Board"
 	select TQC_SDMMC
 	help
@@ -43,6 +43,14 @@ config TQMa8QXS_MB_SMARC_2
 
 endchoice
 
+config DEFAULT_DEVICE_TREE
+	default "fsl-imx8dx-tqma8dxs-mb-smarc-2" if TQMA8QXS_CPU_MX8DX && TQMA8QXS_MB_SMARC_2
+	default "fsl-imx8qxp-tqma8qxps-mb-smarc-2" if TQMA8QXS_CPU_MX8QXP && TQMA8QXS_MB_SMARC_2
+
+config DEFAULT_FDT_FILE
+	default "fsl-imx8dx-tqma8dxs-mb-smarc-2.dtb" if TQMA8QXS_CPU_MX8DX && TQMA8QXS_MB_SMARC_2
+	default "fsl-imx8qxp-tqma8qxps-mb-smarc-2.dtb" if TQMA8QXS_CPU_MX8QXP && TQMA8QXS_MB_SMARC_2
+
 choice
 	prompt "TQMa8QXS boot device"
 	default TQMA8QXS_BOOT_SDCARD
diff --git a/board/tqc/tqma8qxs/Makefile b/board/tqc/tqma8qxs/Makefile
index b03d2d2ea0..bad4cf1ba8 100644
--- a/board/tqc/tqma8qxs/Makefile
+++ b/board/tqc/tqma8qxs/Makefile
@@ -5,5 +5,5 @@
 #
 
 obj-y += tqma8qxs.o
-obj-$(CONFIG_TQMa8QXS_MB_SMARC_2) += tqma8qxs-mb-smarc-2.o
+obj-$(CONFIG_TQMA8QXS_MB_SMARC_2) += tqma8qxs-mb-smarc-2.o
 
diff --git a/include/configs/tqma8qxs.h b/include/configs/tqma8qxs.h
index 829387fd96..408ba66734 100644
--- a/include/configs/tqma8qxs.h
+++ b/include/configs/tqma8qxs.h
@@ -355,7 +355,7 @@
 #define CONFIG_SYS_MEMTEST_SCRATCH CONFIG_SYS_MEMTEST_END
 #endif
 
-#if defined(CONFIG_TQMa8QXS_MB_SMARC_2)
+#if defined(CONFIG_TQMA8QXS_MB_SMARC_2)
 #include "tqma8qxs-mb-smarc-2.h"
 #else
 #error
